test/PushNotificationTests/Shared.cpp

```diff
@@ -0,0 +1,155 @@
+
+        wil::unique_handle process{ ei.hProcess };
+        return process;
+    }
+
+    void RunCertUtil(const std::wstring& path, bool removeCert)
+    {
+        std::wstring action = (removeCert ? L"-delstore" : L"-addstore");
+        std::wstring args{ action + L" TrustedPeople " + path };
+        // VERIFY_ARE_NOT_EQUAL(args, args);
+        auto process = Execute(L"%SystemRoot%\\system32\\certutil.exe",
+            args, g_deploymentDir);
+
+        // Wait for the cer to be installed.
+        auto waitResult = WaitForSingleObject(process.get(), c_phaseTimeout);
+        if (waitResult != WAIT_OBJECT_0)
+        {
+            if (waitResult == WAIT_FAILED) VERIFY_WIN32_SUCCEEDED(GetLastError());
+            VERIFY_ARE_EQUAL(waitResult, WAIT_OBJECT_0);
+        }
+
+        // Make sure the exitcode for the tool is success.
+        DWORD exitCode{};
+        THROW_IF_WIN32_BOOL_FALSE(GetExitCodeProcess(process.get(), &exitCode));
+        VERIFY_ARE_EQUAL(exitCode, 0);
+    }
+
+    void InstallPackage(const std::wstring& packagePath)
+    {
+        // Deploy packaged app to register handler through the manifest.
+        PackageManager manager;
+        IVector<Uri> depPackagePaths;
+        auto result = manager.AddPackageAsync(Uri(packagePath), depPackagePaths,
+            DeploymentOptions::ForceApplicationShutdown).get();
+        auto errorText = result.ErrorText();
+        auto errorCode = result.ExtendedErrorCode();
+        VERIFY_SUCCEEDED(errorCode, errorText.c_str());
+    }
+
+    void UninstallPackage(const std::wstring& packageFullName)
+    {
+        // Deploy packaged app to register handler through the manifest.
+        PackageManager manager;
+        IVector<Uri> depPackagePaths;
+        auto result = manager.RemovePackageAsync(packageFullName).get();
+        auto errorText = result.ErrorText();
+        auto errorCode = result.ExtendedErrorCode();
+        VERIFY_SUCCEEDED(errorCode, errorText.c_str());
+    }
+
+    wil::unique_event CreateTestEvent(const std::wstring& eventName)
+    {
+        bool alreadyExists = false;
+        SECURITY_ATTRIBUTES attributes = {};
+        wil::unique_hlocal_security_descriptor descriptor;
+
+        // Grant access to world and appcontainer.
+        THROW_IF_WIN32_BOOL_FALSE(ConvertStringSecurityDescriptorToSecurityDescriptor(
+            L"D:(A;;GA;;;WD)(A;;GA;;;AC)", SDDL_REVISION_1, &descriptor, nullptr));
+        attributes.nLength = sizeof(SECURITY_ATTRIBUTES);
+        attributes.lpSecurityDescriptor = descriptor.get();
+
+        wil::unique_event event;
+        event.create(wil::EventOptions::None, eventName.c_str(), &attributes, &alreadyExists);
+        return event;
+    }
+
+    void WaitForEvent(const wil::unique_event& successEvent, const wil::unique_event& failedEvent)
+    {
+        HANDLE waitEvents[2] = { successEvent.get(), failedEvent.get() };
+        auto waitResult = WaitForMultipleObjects(_countof(waitEvents), waitEvents, FALSE,
```

Comment:
`_countof` is a MSVC nonstandard extension. Should avoid if standard-conforming alternatives exist. (Like `ARRAYSIZE`, as ugly as it is.)

Reply:
`ARRAYSIZE()` is the recommended syntax for Reunion work (and most anyone else too :-)